File: Windows8Phone/CSharp/Plexi/Interfaces/IPlexiService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Plexi.Models;

namespace Plexi.Interfaces
{
    public interface IPlexiService
    {
        string OriginalQuery { get; }

        void ClearContext();

        void ResetTimer();

        void Query(string query);

        void Choice(ChoiceModel choice);

        event EventHandler onShow;

        event EventHandler onAct;

        event EventHandler onChoice;

        event EventHandler onError;

        event EventHandler onProgress;
    }
}

File: Android/Java/Plexi/Plexi/src/main/java/com/stremor/plexi/interfaces/IPlexiService.java
package com.stremor.plexi.interfaces;

import com.stremor.plexi.models.Choice;

/**
 * Created by jeffschifano on 10/28/13.
 */
public interface IPlexiService {

    public String getOriginalQuery();

    void clearContext();

    void resetTimer();

    void login(String username, String password);

    void signup(String username, String password);

    void query(String query);

    void choice(Choice choice);
}
